app/Models/Extensions/BaseAlbum.php

```diff
@@ -50,6 +50,8 @@ abstract class BaseAlbum extends Model implements AbstractAlbum, HasRandomID
 	 */
 	protected $keyType = HasRandomID::ID_TYPE;
 
+	protected ?string $trackId = null;
```

Comment:
You said, that only `Album` support tracks. So this attribute should not exist on `BaseAlbum`.

Also not that none of the Eloquent DB attributes exist as proper PHP class attributes (something I really hate about Laravel). They are added through PHP magic methods.

Reply:
This was necessary to prevent exceptions that this field (`track_id`) does norexist if requesting it. It will always be `null`. I'll try to look for another way.